File: Mixter.Domain/Core/Messages/Events/MessageDeleted.cs
using Mixter.Domain.Identity;

namespace Mixter.Domain.Core.Messages.Events
{
    [Event]
    public struct MessageDeleted : IDomainEvent
    {
        public MessageId MessageId { get; private set; }

        public UserId Deleter { get; private set; }

        public MessageDeleted(MessageId messageId, UserId deleter) 
            : this()
        {
            MessageId = messageId;
            Deleter = deleter;
        }

        public object GetAggregateId()
        {
            return MessageId;
        }
    }
}

File: src/main/java/mixter/domain/message/events/MessageDeleted.java
package mixter.domain.message.events;

import mixter.Event;
import mixter.domain.message.MessageId;

public class MessageDeleted implements Event {
    private MessageId messageId;

    public MessageDeleted(MessageId messageId) {
        this.messageId = messageId;
    }

    public MessageId getMessageId() {
        return messageId;
    }
}
